File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/bots/BotsInfoRequest.java
package com.github.seratch.jslack.api.methods.request.bots;

import com.github.seratch.jslack.api.methods.SlackApiRequest;

public class BotsInfoRequest implements SlackApiRequest {

    /**
     * Authentication token. Requires scope: `users:read`
     */
    private String token;

    /**
     * Bot user to get info on
     */
    private String bot;

    BotsInfoRequest(String token, String bot) {
        this.token = token;
        this.bot = bot;
    }

    public static BotsInfoRequestBuilder builder() {
        return new BotsInfoRequestBuilder();
    }

    public String getToken() {
        return this.token;
    }

    public String getBot() {
        return this.bot;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public void setBot(String bot) {
        this.bot = bot;
    }

    public static class BotsInfoRequestBuilder {
        private String token;
        private String bot;

        BotsInfoRequestBuilder() {
        }

        public BotsInfoRequest.BotsInfoRequestBuilder token(String token) {
            this.token = token;
            return this;
        }

        public BotsInfoRequest.BotsInfoRequestBuilder bot(String bot) {
            this.bot = bot;
            return this;
        }

        public BotsInfoRequest build() {
            return new BotsInfoRequest(token, bot);
        }

        public String toString() {
            return "BotsInfoRequest.BotsInfoRequestBuilder(token=" + this.token + ", bot=" + this.bot + ")";
        }
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/bots/BotsInfoRequest.kt
package work.socialhub.kslack.api.methods.request.bots

import work.socialhub.kslack.api.methods.FormRequest
import work.socialhub.kslack.api.methods.SlackApiRequest

class BotsInfoRequest(
    /** Authentication token. Requires scope: `users:read` */
    override var token: String?,
    /** Bot user to get info on */
    var bot: String?
) : SlackApiRequest, FormRequest {

    override fun toMap(): Map<String, Any> {
        return mutableMapOf<String, Any>().also {
            it.addParam("bot", bot)
        }
    }
}